Two minor linting adjustments

lennrt opened the pull request:
-  `cmd/client_test.go`,  `main.go`, and `main_test.go` had a few instances of unkeyed fields in composite literals. This is a non-issue if the `InstanceAppFactory` and `TerminalAuthenticator` structs are never expanded, but if additional fields are ever added to the structs then unkeyed initilizations may break.
 
- I noticed that the method receivers for `Create(c *gin.Context)` and `Show(c *gin.Context)` in `web/jobs_controller.go` were different from the receiver for `Index(c *gin.Context)`, despite referring to the same `JobsController` struct. I've adjusted these for the sake of consistency.

dimroc commented on web/jobs_controller.go:
```diff
@@ -35,18 +35,18 @@ func (jrc *JobsController) Index(c *gin.Context) {
 // Example:
 //  "<application>/jobs"
-func (jc *JobsController) Create(c *gin.Context) {
+func (jrc *JobsController) Create(c *gin.Context) {
```
`jc` is an abbreviation of `JobsController`.
`jrc` is an abbreviation of `JobRunsController`.

It looks like you should change `Index` to have a `jc *JobsController` receiver rather than this.

dimroc commented:
Thanks for the PR @lennrt. What are you using for linting? `golint`?

lennrt commented:
Hi @dimroc, yes, I use `golint` but I also use [`go vet`](https://golang.org/cmd/vet/) to find additional things not caught by the compiler and `golint`. 

There were a few other changes suggested by `golint`, but they are mainly stylistic and  are unlikely to ever lead to any errors. Let me know if you'd like me to make any of these changes as well:

```
../src/chainlink/adapters/http.go:14:6: type HttpGet should be HTTPGet
../src/chainlink/adapters/http.go:42:6: type HttpPost should be HTTPPost
../src/chainlink/adapters/json_parse.go:15:6: type JsonParse should be JSONParse
../src/chainlink/services/subscription.go:83:6: type RpcLogSubscription should be RPCLogSubscription
../src/chainlink/services/subscription.go:94:6: func NewRpcLogSubscription should be NewRPCLogSubscription
../src/chainlink/services/subscription.go:200:6: type RpcLogEvent should be RPCLogEvent
```

dimroc commented:
Thanks @lennrt, all of those look like things we would like. Feel free to make those changes.

lennrt commented:
- I made the following naming changes: HttpGet -> HTTPGet; HttpPost -> HTTPPost; JsonParse -> JSONParse; RpcLogSubscription -> RPCLogSubscription; NewRpcLogSubscription -> NewRPCLogSubscription; RpcLogEvent -> RPCLogEvent.
 
- In doing this I also noticed that in https://github.com/smartcontractkit/chainlink/blob/88e9047fc572c14279e6df0184d0ee3e8b2f7a8d/adapters/http_test.go#L18 the HTTPPost adapter test case was using a `Get` adapter rather than a `Post` adapter. I changed this to a `Post` adapter.

- I also noticed that in https://github.com/smartcontractkit/chainlink/blob/88e9047fc572c14279e6df0184d0ee3e8b2f7a8d/adapters/http.go#L48 the method receiver was named `hga *HTTPPost`. I changed this to `(hpa *HTTPPost)` to correctly reflect the fact that the receiver is a `Post` receiver rather than a `Get` receiver.

dimroc commented:
Thanks a lot @lennrt 👍
